tracer/test/Datadog.Trace.Tests/Logging/DirectSubmission/Sink/OtlpSinkTests.cs

```diff
@@ -0,0 +1,291 @@
+// <copyright file="OtlpSinkTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#if NETCOREAPP3_1_OR_GREATER
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Datadog.Trace.ClrProfiler.AutoInstrumentation.Logging.ILogger.DirectSubmission;
+using Datadog.Trace.Logging.DirectSubmission.Sink;
+using Datadog.Trace.Logging.DirectSubmission.Sink.PeriodicBatching;
+using Datadog.Trace.OpenTelemetry;
+using Datadog.Trace.OpenTelemetry.Logs;
+using FluentAssertions;
+using Xunit;
+
+namespace Datadog.Trace.Tests.Logging.DirectSubmission.Sink
+{
+    public class OtlpSinkTests
+    {
+        private const int DefaultQueueLimit = 100_000;
+        private static readonly TimeSpan TinyWait = TimeSpan.FromMilliseconds(50);
+
+        [Fact]
+        public void SinkSendsLogsToOtlpExporter()
+        {
+            var mutex = new ManualResetEventSlim();
+            var capturedLogs = new List<LogPoint>();
+
+            var options = new BatchingSinkOptions(batchSizeLimit: 2, queueLimit: DefaultQueueLimit, period: TinyWait);
+            var sink = new TestOtlpSink(options, logs =>
```

Comment:
Rather than create a new sink to test, I think the most effective way to test this would be to use the exact same class that is going to be used at runtime, which is the `OtlpSubmissionLogSink` object. Perhaps you can pass an `OtlpExporter` into the `OtlpSubmissionLogSink` constructor, so this test code could be controlled by creating a test `OtlpExporter` class rather than a test `OtlpSubmissionLogSink` class